### PR TITLE
Support for input lists coercion on `Variables` types

### PR DESCRIPTION
**Pull request by gilgardosh**
related: [#4888](https://github.com/dotansimha/graphql-code-generator/issues/4888)

### CODE REVIEW

File: dev-test/test-schema/resolvers-root.ts [hunk 1]
```diff
@@ -150,14 +150,14 @@ export type QueryRootResolvers<
   ContextType = any,
   ParentType extends ResolversParentTypes['QueryRoot'] = ResolversParentTypes['QueryRoot']
 > = {
-  allUsers?: Resolver<Array<Maybe<ResolversTypes['User']>>, ParentType, ContextType>;
+  allUsers?: Resolver<Array<Maybe<ResolversTypes['User']>> | Maybe<ResolversTypes['User']>, ParentType, ContextType>;
```
Comment (dotansimha) [inline dev-test/test-schema/resolvers-root.ts:153]: This should be effected here? I'm not sure.

**Review by dotansimha — CHANGES_REQUESTED** [review]

File: dev-test/test-schema/flow-types.flow.js [hunk 2]
```diff
@@ -111,7 +111,11 @@ export type ResolversParentTypes = {
 };
 
 export type QueryResolvers<ContextType = any, ParentType = $ElementType<ResolversParentTypes, 'Query'>> = {
-  allUsers?: Resolver<Array<?$ElementType<ResolversTypes, 'User'>>, ParentType, ContextType>,
+  allUsers?: Resolver<
```
Comment (dotansimha) [inline dev-test/test-schema/flow-types.flow.js:114]: Shouldn't get modified as well, right?

File: packages/plugins/flow/flow/tests/__snapshots__/flow.spec.ts.snap [hunk 3]
```diff
@@ -126,9 +126,9 @@ export type MyType = {|
 export type MyTypeFooArgs = {|
```
Comment (dotansimha) [inline packages/plugins/flow/flow/tests/__snapshots__/flow.spec.ts.snap:126]: I'm not 100% sure that we should modify `Args`. I think only `variables` should be effected. 

Let's say you have an argument defined for field, if you define it as `[String]`, your Args will always be `string[]`. But, on the level of consumption, where you use variables, you should be able to use `string | string[]`. 

What do you think?

**Review by dotansimha — CHANGES_REQUESTED** [review]

File: dev-test/test-schema/flow-types.flow.js (hunk 2 above)
Comment (gilgardosh) [reply dev-test/test-schema/flow-types.flow.js]: It shouldn't. same issue accured on TS resolvers.
Fixed.

File: packages/plugins/flow/flow/tests/__snapshots__/flow.spec.ts.snap (hunk 3 above)
Comment (gilgardosh) [reply packages/plugins/flow/flow/tests/__snapshots__/flow.spec.ts.snap:126]: I actually don't know if GraphQL support coercion on `Args`
we should try :)

Comment (dotansimha) [reply packages/plugins/flow/flow/tests/__snapshots__/flow.spec.ts.snap:126]: Let's just apply it for `Variables` for now. I think `Args` will always be the way you define it.

File: packages/plugins/typescript/mongodb/src/visitor.ts [hunk 4]
```diff
@@ -23,6 +22,7 @@ import {
   InterfaceTypeDefinitionNode,
   UnionTypeDefinitionNode,
 } from 'graphql';
+import { wrapTypeNodeWithModifiers } from '../../../other/visitor-plugin-common/src/utils';
```
Comment (dotansimha) [inline packages/plugins/typescript/mongodb/src/visitor.ts:25]: Please fix this import, it should take it from `@graphql-codegen/visitor-plugin-common`

File: .changeset/fresh-balloons-talk.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (dotansimha) [inline .changeset/fresh-balloons-talk.md:1]: Please update your `changeset`, since `flow-operations` has also changed.

Comment (dotansimha) [reply .changeset/fresh-balloons-talk.md:1]: Also, `visitor-plugin-common` has changed.

**Review by dotansimha — CHANGES_REQUESTED** [review]

**Pull request merged**